src/libasr/codegen/llvm_utils.cpp

```diff
@@ -1962,6 +1962,21 @@ namespace LCompilers {
                 }
                 return tuple_hash;
             }
+            case ASR::ttypeType::Logical: {
+                llvm::AllocaInst *hash_value = builder->CreateAlloca(llvm::Type::getInt32Ty(context), nullptr);
+
+                llvm::Value* is_key_one = builder->CreateICmpEQ(key,
+                    llvm::ConstantInt::get(context, llvm::APInt(1, 1)));
+                
+                llvm_utils->create_if_else(is_key_one, [&]() {
+                    LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 1)), hash_value);
+                }, [=]() {
+                    LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), hash_value);
+                });
+                return hash_value;
+            }
```

Comment:
This logic is correct but the if-else check seems unnecessary. A simple cast from `i1` to `i32` should have done the job. We will see in our tomorrow's meeting how can we do that.